nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/GetMongo.java

```diff
@@ -151,17 +172,34 @@ public ValidationResult validate(final String subject, final String value, final
         return propertyDescriptors;
     }
 
-    private ObjectMapper mapper = new ObjectMapper();
+    private ObjectMapper mapper;
 
     //Turn a list of Mongo result documents into a String representation of a JSON array
-    private String buildBatch(List<Document> documents) throws IOException {
-        List<Map> docs = new ArrayList<>();
-        for (Document document : documents) {
-            String asJson = document.toJson();
-            docs.add(mapper.readValue(asJson, Map.class));
+    private String buildBatch(List<Document> documents, String jsonTypeSetting) throws IOException {
+        StringBuilder builder = new StringBuilder();
+        for (int index = 0; index < documents.size(); index++) {
+            Document document = documents.get(index);
+            String asJson;
+            if (jsonTypeSetting.equals(JSON_TYPE_NORMAL)) {
+                asJson = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(document);
```

Comment:
Should there be a third JSON Type option of ("Standard Pretty-Printed JSON")? For speed and/or size concerns, we wouldn't want to force pretty-printing, but it might be nice as an option. An additional property is probably overkill, especially if the Extended JSON output isn't pretty-printed, but if they could each be pretty printed, then perhaps an additional property (defaulted to false) is the right way to go.